pkg/svcat/service-catalog/plan.go

```diff
@@ -55,44 +56,69 @@ type Plan interface {
 	// GetDescription returns the plan description.
 	GetDescription() string
 
+	// GetFree returns if the plan is free.
+	GetFree() bool
+
 	// GetClassID returns the plan's class name.
 	GetClassID() string
+
+	// GetInstanceCreateSchema returns the instance create schema from plan.
+	GetInstanceCreateSchema() *runtime.RawExtension
+
+	// GetInstanceUpdateSchema returns the instance update schema from plan.
+	GetInstanceUpdateSchema() *runtime.RawExtension
+
+	// GetBindingCreateSchema returns the instance create schema from plan.
+	GetBindingCreateSchema() *runtime.RawExtension
 }
 
 // RetrievePlans lists all plans defined in the cluster.
-func (sdk *SDK) RetrievePlans(opts RetrievePlanOptions) ([]Plan, error) {
+func (sdk *SDK) RetrievePlans(classID string, opts ScopeOptions) ([]Plan, error) {
+	plans, err := sdk.retrievePlansByListOptions(opts, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if classID == "" {
```

Comment:
I think that `spec.clusterServiceClassRef.name` or `spec.serviceClassRef.name` can finally be specified as field selectors when retrieving a list of plans from the API. As a follow-up, it would be great to switch from client-side filtering to filtering in the list options.